lib/features/contact/presentation/pages/contacts_page.dart

```diff
@@ -1,118 +1,69 @@
+import 'package:chat_app/features/chat/data/repositories/chat_repository_impl.dart';
+import 'package:chat_app/features/contact/data/data_sources/contact_local_data_source.dart';
+import 'package:chat_app/features/contact/data/data_sources/contact_remote_data_source.dart';
+import 'package:chat_app/features/contact/data/repositories/contact_repository_impl.dart';
+import 'package:chat_app/features/contact/domain/use_cases/add_contact_use_case.dart';
+import 'package:chat_app/features/contact/domain/use_cases/load_contacts_use_case.dart';
+import 'package:chat_app/features/contact/domain/use_cases/load_or_create_chat_conversation_use_case.dart';
 import 'package:chat_app/features/contact/presentation/bloc/contact_bloc.dart';
+import 'package:chat_app/features/contact/presentation/views/contacts_view.dart';
+import 'package:cloud_firestore/cloud_firestore.dart';
 import 'package:firebase_auth/firebase_auth.dart';
 import 'package:flutter/material.dart';
 import 'package:flutter_bloc/flutter_bloc.dart';
-import 'package:flutter_svg/flutter_svg.dart';
-import 'package:go_router/go_router.dart';
+import 'package:get_it/get_it.dart';
 
 class ContactsPage extends StatelessWidget {
-  ContactsPage({super.key});
-
-  final TextEditingController searchController = TextEditingController();
+  const ContactsPage({super.key});
 
   @override
   Widget build(BuildContext context) {
-    return Scaffold(
-      appBar: AppBar(
-        title: const Text(
-          'Contacts',
-          style: TextStyle(
-            fontSize: 24,
-            fontWeight: FontWeight.w500,
-          ),
-        ),
-        actions: [
-          const SizedBox.shrink(),
-          IconButton(
-            onPressed: () {},
-            icon: SvgPicture.asset(
-              addContact,
-            ),
+    final sharedPreferencesHelper = GetIt.instance<SharedPreferencesHelper>();
+    final fireBaseAuth = GetIt.instance<FirebaseAuth>();
+    final firebaseFirestore = GetIt.instance<FirebaseFirestore>();
+
+    //Contact Repository
+    final contactLocalDataSource =
+        ContactLocalDataSource(sharedPreferencesHelper);
+    final contactRemoteDataSource = ContactRemoteDataSource(firebaseFirestore);
+
+    final contactRepository = ContactRepositoryImpl(
+      remoteDataSource: contactRemoteDataSource,
+      localDataSource: contactLocalDataSource,
+    );
+
+    //Chat Repository
+    final chatLocalDataSource = ChatLocalDataSource(sharedPreferencesHelper);
+    final chatRemoteDataSource = ChatRemoteDataSource(firebaseFirestore);
+
+    final chatRepository = ChatRepositoryImpl(
+      remoteDataSource: chatRemoteDataSource,
+      localDataSource: chatLocalDataSource,
+    );
+
+    //UseCases
+    final addContactUseCase = AddContactUseCase(contactRepository);
+    final loadContactsUseCase = LoadContactsUseCase(contactRepository);
+    final loadOrCreateChatConversationUseCase =
+        LoadOrCreateChatConversationUseCase(
+      chatRepository,
+    );
+
```

Comment:
you are not suppose to construct these out of providers. that is the whole point of lazy loading. build function should never initiate any object.